common/lib/client/channel.ts

```diff
@@ -189,6 +190,17 @@ class Channel extends EventEmitter {
   _publish(requestBody: unknown, headers: Record<string, string>, params: any, callback: ResourceCallback): void {
     Resource.post(this.rest, this.basePath + '/messages', requestBody, headers, params, null, callback);
   }
+
+  status(callback?: StandardCallback<API.Types.ChannelDetails>): void | Promise<API.Types.ChannelDetails> {
+    if (typeof callback !== 'function' && this.rest.options.promises) {
+      return Utils.promisify(this, 'status', []);
+    }
+
+    const format = this.rest.options.useBinaryProtocol ? Utils.Format.msgpack : Utils.Format.json;
+    const headers = Utils.defaultPostHeaders(format);
```

Comment:
Do we need to pass any headers? I think it works fine without passing any headers -> 
https://github.com/ably/ably-go/pull/521

Reply:
We definitely need to be sending the `Ably-Agent` and `X-Ably-Version` headers with REST API requests in all of our SDKs. Requests will still work fine without them but it would stop us from being able to track those requests in our metrics.